### PR TITLE
Fix Traffic Monitor to startup when cache bytes don't change

### PR DESCRIPTION
**Pull request by rob05c**
This fixes an issue where TM refused to start if a cache's bytes in
astats never change. TM returns 503 until it's polled every cache
twice, and has a BytesPerSecond calculation. The issue was, if a cache
had hard-coded proc.net.dev which never changed, TM has no way to know
the cache simply hasn't updated its stats, and that we won't get a
different stat on the next poll.

This gives TM a timeout, once it successfully polls a cache once,
if the stats don't change in 60s, we start up anyway. It's not
reasonable for a cache to update stats less frequently than a minute.
While guessing isn't ideal, there's no good alternative. We can't
know the cache simply hasn't updated its stats internally; the
only option is a timeout, or never starting.

Incidentally, this problem occurred in production. We had a bad
cache lying to us with hard-coded astats that never updated, causing
TM to never start.

### CODE REVIEW

File: traffic_monitor_golang/traffic_monitor/threadsafe/polledcaches.go [hunk 1]
```diff
@@ -103,11 +116,22 @@ func copyCaches(a map[enum.CacheName]struct{}) map[enum.CacheName]struct{} {
...
 
+func copyCachesTime(a map[enum.CacheName]time.Time) map[enum.CacheName]time.Time {
+	b := map[enum.CacheName]time.Time{}
+	for k, v := range a {
+		b[k] = v
+	}
+	return b
+}
+
+const PolledBytesPerSecTimeout = time.Second * 60
```
Comment (mtorluemke) [inline traffic_monitor_golang/traffic_monitor/threadsafe/polledcaches.go]: Good concept -- I would pick something closer to 10 seconds, though.

Comment (dneuman64) [inline traffic_monitor_golang/traffic_monitor/threadsafe/polledcaches.go]: Can we make it a config?

Comment (rob05c) [inline traffic_monitor_golang/traffic_monitor/threadsafe/polledcaches.go]: It makes me anxious that if we had trouble communicating with a cache which was overloaded with traffic for 10s, we'd tell the Router it's online and has 0 bytes/second and to send it lots of traffic.

It'd take a bit more work to make it a config, that object in the code has no access to the config right now. I'm not a big fan of adding configs for every possible thing, though, every config option is more ops work and training. One doesn't seem like a big deal, but they add up. This should be a rare case; it just happens that we've allowed people to put a dysfunctional 'cache' on our CDN.

Comment (mtorluemke) [inline traffic_monitor_golang/traffic_monitor/threadsafe/polledcaches.go]: "Trouble communicating with a cache" should, by itself, mark the cache as unhealthy.

**Comment by asfgit** [comment]
Can one of the admins verify this patch?

File: traffic_monitor_golang/traffic_monitor/threadsafe/polledcaches.go (hunk 1 above)
Comment (rob05c) [inline traffic_monitor_golang/traffic_monitor/threadsafe/polledcaches.go]: Changed.

**Comment by asfgit** [comment]
Can one of the admins verify this patch?

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/incubator-trafficcontrol-traffic_ops-test-PR/12/

**Comment by asfgit** [comment]
Refer to this link for build results (access rights to CI server needed): 
https://builds.apache.org/job/incubator-trafficcontrol-PR-trafficops-test/12/

**Pull request merged**